Sources/Commands/SwiftPackageTool.swift

```diff
@@ -921,17 +922,25 @@ extension SwiftPackageTool {
             let plugin = matchingPlugins[0]
             swiftTool.observabilityScope.emit(info: "Running plugin \(plugin)")
             
-            // Find the targets (if any) specified by the user. We expect them in the root package.
+            // The proposal calls for applying plugins to regular targets.
+            func isEligible(target: ResolvedTarget) -> Bool {
+                return ![.plugin, .snippet, .systemModule].contains(target.type)
+            }
```

Comment:
can this be made into an extension on ResolvedTarget?

Reply:
👍